persistence-modules/spring-boot-persistence-2/src/main/java/com/baeldung/db/indexing/FileSystemRepository.java

```diff
@@ -0,0 +1,36 @@
+package com.baeldung.db.indexing;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Date;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+interface FileSystemRepository extends JpaRepository<Image, Long> {
+
+    String RESOURCES_DIR = FileSystemRepository.class.getResource("/")
+        .getPath() + "/";
+
+    default Image saveToFileSystem(Image image) throws Exception {
+        Path newFile = Paths.get(RESOURCES_DIR + new Date().getTime() + "-" + image.name);
+        Files.createDirectories(newFile.getParent());
+
+        Files.write(newFile, image.content);
+
+        image.setName(image.name);
+        image.setLocation(newFile.toAbsolutePath()
+            .toString());
+        image.setContent(null);
```

Comment:
On the whole, I think it would make cleaner code if the image you saved was a different object than the input. This works, which is fine, but it's kind of dirty as the caller provides you an image with its bytes and you delete them.

Better this should construct `Image imageToSave = new Image(image.getName(), newFile.toAbsolutePath());`

Also note - constructors are better than setters!